src/internal/account.ts

```diff
@@ -169,6 +169,47 @@ export async function getResource(args: {
   return data;
 }
 
+export async function lookupOriginalAccountAddress(args: {
+  aptosConfig: AptosConfig;
+  addressOrAuthKey: HexInput;
+  options?: LedgerVersion;
+}): Promise<AccountAddress> {
```

Comment:
can't we just return the string instead of converting it into AccountAddress?

Reply:
Hmm, personally I prefer AccountAddress since it follows OOD better. Similar to the Account.Sign method, which returns the Signature instead of hex data. 

With the AccountAddress, user can chose to convert to String (with or without prefix) or U8intArray easier.

Reply:
hmm I think it is a bit different, the query response is just the address, then we might want to return whatever the server returns....

Reply:
Or I can have the this query to return a raw string, but having `lookupOriginalAccountAddress` method from the api/account to return `AccountAddress`?

Reply:
I generally agree with returning `AccountAddress`, especially since we're now stringently parsing everything and it should all be kosher in terms of its string representation with `.toString()`